### PR TITLE
[CUR-301] Added Avo tracking to curric pages 

### PR DESCRIPTION
**Pull request by sonalisin**
## Description

- Added the following tracking events:

### Unit Tab
- Curriculum Thread Highlighted
- Year Group Selected 

### Unit Modal
- Unit modal is opened

### Subject Phase Picker
- Curriculum Visualiser is accessed through subject phase picker

## How to test

1. Go to https://deploy-preview-2039--oak-web-application.netlify.thenational.academy
2. Click on the curriculum visualiser
3. You should see no change in behaviour to any of the above events
4. If using the console, you should be able to see events sent through like below 

Screenshot 2023-10-24 at 11 17 23

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/pages/CurriculumInfo/CurriculumHeader/CurriculumHeader.test.tsx [hunk 1]
```diff
@@ -1,11 +1,17 @@
 import CurriculumHeader from "./CurriculumHeader";
 
-import renderWithTheme from "@/__tests__/__helpers__/renderWithTheme";
+// import render from "@/__tests__/__helpers__/render";
```
Comment (njblore) [inline src/components/pages/CurriculumInfo/CurriculumHeader/CurriculumHeader.test.tsx:3]: 👁️

File: src/components/UnitModal/UnitModal.tsx [hunk 2]
```diff
@@ -47,11 +53,31 @@ const UnitModal: FC<UnitModalProps> = ({
...
       setUnitOptionsAvailable(false);
     }
   }, [displayModal, setUnitOptionsAvailable, optionalityModalOpen]);
 
+  useEffect(() => {
+    // For tracking open model events
+    if (displayModal === true) {
+      if (unitData) {
+        track.unitInformationViewed({
+          unitName: unitData.title /* string */,
```
[the remark this answers is not in the document]
Comment (sonalisin) [reply src/components/UnitModal/UnitModal.tsx]: Ah it was codegen I copied and pasted in! The types are already declared in Avo so I'll remove them...

File: src/__tests__/pages/teachers/curriculum/[subjectPhaseSlug]/[tab].test.tsx [hunk 3]
```diff
@@ -53,6 +59,9 @@ jest.mock("@/pages/teachers/curriculum/index", () => ({
 }));
 
 describe("pages/teachers/curriculum/[subjectPhaseSlug]/[tab]", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
```
[the remark this answers is not in the document]
Comment (sonalisin) [reply src/__tests__/pages/teachers/curriculum/[subjectPhaseSlug]/[tab].test.tsx:63]: I wanted to keep it consistent with all the other tests which use `beforeEach()`. It definitely wouldn't hurt to have an `afterEach` though also

**Review by njblore — APPROVED** [review]

**Pull request merged**